### PR TITLE
feat: add provider checklist

### PR DESCRIPTION
**Pull request by toddbaert**
- Adds provider checklist and FAQ
- Adds md checklist component that allows you to copy rendered checklist as raw markdown.

see: https://deploy-preview-96--lucky-creponne-baf9b3.netlify.app/docs/reference/concepts/provider#checklist

### CODE REVIEW

File: docs/reference/concepts/02-provider.mdx [hunk 1]
```diff
@@ -246,3 +248,63 @@ func (e MyFeatureProvider) Hooks() []Hook {
...
+#### How should I handle error conditions?
+
+If an error condition is encountered in your provider, an error should be [indicated](/docs/specification/sections/providers#flag-value-resolution) by throwing or returning an error, as language idioms dictate. The SDK will ensure the default value is returned and expose the relevant error data.
+
+#### My flag system doesn't support flags of a particular type. What should I do?
+
+If your backend system doesn't support a particular flag type, you can:
+
+- Do a conversion from a type the system supports (for example, store JSON objects in a string, or booleans as `0`/`1`), and document this pattern.
+- Log and indicate an error in your provider (this will result in the default value being returned).
```
Comment (agentgonzo) [inline docs/reference/concepts/02-provider.mdx:298]: I would suggest changing this to just 'throw an error'. Log-and-throw is an antipattern that end up with multiple versions of the same problem being logged by different layers of the code.

[Having gone through this in a provider](https://github.com/rollout/cloudbees-openfeature-provider-java/blob/main/src/main/java/com/cloudbees/openfeature/provider/ContextTransformer.java#L46) WIP, it is neater (from a provider's point of view) just to throw an error with the reason embedded in it. 

IMO the OpenFeature SDK should do the logging of the error.

File: docs/reference/concepts/02-provider.mdx [hunk 2]
```diff
@@ -246,3 +248,63 @@ func (e MyFeatureProvider) Hooks() []Hook {
...
+
+#### How can people find my provider?
+
+Please add your provider to the [documentation](https://github.com/open-feature/docs.openfeature.dev/issues/new?assignees=&labels=hook&template=document-hook.yaml&title=%5BHook%5D%3A+) so people can find it!
+
+#### Where should I put my provider source code?
+
+If you are a vendor and you want to maintain full control over your provider, you should host the provider in your organization's SCM. If you would like your provider to be maintained as a community project, please take a look at our various [contrib repositories](https://github.com/orgs/open-feature/repositories?q=contrib&type=all&language=&sort=).
```
Comment (agentgonzo) [inline docs/reference/concepts/02-provider.mdx:306]:
```suggestion
If you are a vendor and you want to maintain full control over your provider, you should host the provider in your organization's SCM. If you would like your provider to be maintained as a community project, please take a look at our various [contrib repositories](https://github.com/orgs/open-feature/repositories?q=contrib&type=all&language=&sort=). We encourage you to make your code open source, but obviously it's your decision.
```

File: docs/reference/concepts/02-provider.mdx (hunk 1 above)
Comment (toddbaert) [reply docs/reference/concepts/02-provider.mdx:298]: You're right.

File: docs/reference/concepts/02-provider.mdx [hunk 3]
```diff
@@ -246,3 +248,63 @@ func (e MyFeatureProvider) Hooks() []Hook {
...
+Here is a checklist of important considerations when implementing a provider.
+
+#### Functional requirements
+
+<CopyChecklist items={[
+  'All resolution methods are implemented (see `Provider` / `FeatureProvider` interface in applicable SDK).',
+  'Relevant configuration options for the underlying backend are exposed (typically using the provider constructor).',
+  'Flag-keys are transformed appropriately for the underlying backend (OpenFeature defines no restrictions in valid flag keys).',
+  'Context is transformed appropriately for the underlying backend.',
+  'Provider is named according to the [naming conventions](#naming_recommendations)'
```
Comment (beeme1mr) [inline docs/reference/concepts/02-provider.mdx:263]: Full URL

File: docs/reference/concepts/02-provider.mdx [hunk 4]
```diff
@@ -246,3 +248,63 @@ func (e MyFeatureProvider) Hooks() []Hook {
...
+]} />
+
+#### Non-functional requirements
+
+<CopyChecklist items={[
+  'Documentation for installation, all behavior (options, context transformation, flag-key transformation).',
+  'Provider is adequately tested.',
+  'Provider is added to [the documentation site](https://github.com/open-feature/docs.openfeature.dev/issues/new?assignees=&labels=hook&template=document-hook.yaml&title=%5BHook%5D%3A+).',
+  'Provider releases follow [semantic versioning](https://semver.org/).',
+  'Document the version of the OpenFeature specification to which your provider complies. '
```
Comment (beeme1mr) [inline docs/reference/concepts/02-provider.mdx:273]:
```suggestion
  'Document the version of the OpenFeature specification to which your provider complies.'
```

File: docs/reference/concepts/02-provider.mdx [hunk 5]
```diff
@@ -246,3 +248,63 @@ func (e MyFeatureProvider) Hooks() []Hook {
...
+
+#### Does my provider's version need to match the spec version to which it complies?
+
+No. You should document the version of the OpenFeature specification your provider complies to, and otherwise use [semantic versioning](https://semver.org/) in your artifact numbers.
+
+#### My backend need a targeting key (or other user identifier), how can I ensure one exists?
+
+Targeting keys are not required by the Evaluation API. If your backend requires one, you have a couple options:
+
+- Indicate an error in your provider (this will result in the default value being returned).
```
Comment (beeme1mr) [inline docs/reference/concepts/02-provider.mdx:286]: Perhaps add a bullet point item to add the requirement to the provider's readme.

File: docs/reference/concepts/02-provider.mdx [hunk 6]
```diff
@@ -246,3 +248,63 @@ func (e MyFeatureProvider) Hooks() []Hook {
...
+#### My flag system doesn't support flags of a particular type. What should I do?
+
+If your backend system doesn't support a particular flag type, you can:
+
+- Do a conversion from a type the system supports (for example, store JSON objects in a string, or booleans as `0`/`1`), and document this pattern.
+- Indicate an error in your provider (this will result in the default value being returned).
+
+#### How can people find my provider?
+
+Please add your provider to the [documentation](https://github.com/open-feature/docs.openfeature.dev/issues/new?assignees=&labels=hook&template=document-hook.yaml&title=%5BHook%5D%3A+) so people can find it!
```
Comment (beeme1mr) [inline docs/reference/concepts/02-provider.mdx]: This is using the wrong URL. Please use the Provider template.

Comment (toddbaert) [reply docs/reference/concepts/02-provider.mdx]: You are commenting on an our of date commit.

File: docs/reference/concepts/02-provider.mdx (hunk 5 above)
Comment (toddbaert) [reply docs/reference/concepts/02-provider.mdx:286]: Added "Be sure to document this behavior."

File: docs/reference/concepts/02-provider.mdx [hunk 7]
```diff
@@ -246,3 +248,65 @@ func (e MyFeatureProvider) Hooks() []Hook {
...
+  'Document the version of the OpenFeature specification to which your provider complies.'
+]} />
+
+### FAQ
+
+#### Does my provider's version need to match the spec version to which it complies?
+
+No. You should document the version of the OpenFeature specification your provider complies to, and otherwise use [semantic versioning](https://semver.org/) in your artifact numbers.
+
+#### My backend need a targeting key (or other user identifier), how can I ensure one exists?
```
Comment (justinabrahms) [inline docs/reference/concepts/02-provider.mdx:282]:
```suggestion
#### My backend needs a targeting key (or other user identifier), how can I ensure one exists?
```

**Review by justinabrahms — APPROVED** [review]

**Review by beeme1mr — APPROVED** [review]

**Pull request merged**